chatheads/src/main/ets/components/ChatHeads/ChatHeads.ets

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2022 Application Library Engineering Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ChatheadModel, Point, Status, vpPoint } from './Types'
+import { step } from './Step'
+
+@Entry
+@Component
+  /**
+   * counter is used to produce delay in successive chatheads.
+   */
+export struct ChatHead {
+  chatheadModel: ChatheadModel= new ChatheadModel();
+  @State private cPosition: Point = {};
+  @State private cStatus: Array<Status> = [];
+  private counter: number;
+
+  /**
+   * redefines the getPosition function.
+   * initializes counter and current position of all chatheds.
+   */
+  aboutToAppear() {
+    this.chatheadModel.images.length = this.chatheadModel.images.length
```

Comment:
this.chatheadModel.images.length = this.chatheadModel.images.length??
recheck this statement